### ISSUE TITLE
Style/BlockDelimiters forces do-end style even if it were one liner

### ISSUE THREAD

**Issue by ym19851201** [rubocop/rubocop#11117]
I found a bug related to Style/BlockDelimters

--------

## Expected behavior
I hoped Rubocop didn't offend the code below.
```ruby
def method
  ary&.map { |e| some_func(e) }&.find { |e| some_func2(e) }
end
```

## Actual behavior

But actually, it's overwritten by RuboCop like below.
```ruby
def method
  ary&.map do |e| some_func(e) end&.find { |e| some_func2(e) }
end
```

And `rubocop --debug` said 
```
For /current/directory: configuration from /path/to/.rubocop.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
```

## Steps to reproduce the problem
1. Set `Style/BlockDelimiters` EnforcedStyle: semantic
2. Chain some methods with at least one safe navigation operator
  Probably, it should be in a method. On the top level, it seems another kind of bug occurs against method chain. (It's under investigation now)
3. Run RuboCop
4. All pairs of `{}` in the chain except last one will be offended

## RuboCop version

```
$ [bundle exec] rubocop -V
1.37.1 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.2) [arm64-darwin21]
```
